packages/menu/src/SubMenu/SubMenu.spec.tsx

```diff
@@ -8,104 +8,107 @@ import userEvent from '@testing-library/user-event';
 
 import { Menu, MenuItem, SubMenu } from '..';
 
-const SubMenuTestWrapper = ({ children }: { children: React.ReactNode }) => {
-  return (
-    <Menu open={true} setOpen={jest.fn()}>
-      {children}
-      <SubMenu title="Second SubMenu" data-testid="sub-menu-b">
-        <MenuItem data-testid="sub-menu-item-b">SubMenu Item Two</MenuItem>
+const subMenu1Id = 'sub-menu-1-id';
+const subMenu2Id = 'sub-menu-2-id';
+const subMenu3Id = 'sub-menu-3-id';
+const menuItem1Id = 'menu-item-1-id';
+const menuItem2Id = 'menu-item-2-id';
+const onClick = jest.fn();
+
+const renderSubMenu = () => {
+  const utils = render(
+    <Menu>
+      <SubMenu active onClick={onClick} data-testid={subMenu1Id}>
+        <MenuItem data-testid={menuItem1Id}>Text Content A</MenuItem>
       </SubMenu>
-    </Menu>
+      <SubMenu data-testid={subMenu2Id} href="mongodb.design">
+        <MenuItem data-testid={menuItem2Id}> Text Content B</MenuItem>
+      </SubMenu>
+      <SubMenu data-testid={subMenu3Id} as="div"></SubMenu>
+    </Menu>,
   );
+
+  return utils;
 };
 
-describe('packages/menu/sub-menu', () => {
+describe('packages/sub-menu', () => {
   test('renders a SubMenu open by default, when the SubMenu is active', () => {
-    const { getByTestId } = render(
-      <SubMenuTestWrapper>
-        <SubMenu data-testid="sub-menu-item-a" active={true} />
-      </SubMenuTestWrapper>,
-    );
+    const { getByTestId } = renderSubMenu();
 
     // TODO: Fix redundant rendering in `Menu`. The submenu is closed on initial render, but opens on second render
     // https://jira.mongodb.org/browse/LG-2904
     waitFor(() => {
-      const subMenuItem = getByTestId('sub-menu-item-a');
-      expect(subMenuItem).toBeInTheDocument();
+      const subMenu = getByTestId(subMenu1Id);
+      expect(subMenu).toBeInTheDocument();
+      const menuItem = getByTestId(menuItem1Id);
+      expect(menuItem).toBeInTheDocument();
     });
   });
 
   test('when a SubMenu is clicked, it opens and closes the previously opened SubMenu', async () => {
-    const { queryByTestId, getAllByTestId } = render(
-      <SubMenuTestWrapper>
-        <SubMenu data-testid="sub-menu-a" active={true}>
-          <MenuItem data-testid="sub-menu-item-a">SubMenu Item One</MenuItem>
-        </SubMenu>
-      </SubMenuTestWrapper>,
-    );
-    const [subMenuButtonA, subMenuButtonB] = getAllByTestId(
-      'lg-sub-menu-icon-button',
-    );
-
-    userEvent.click(subMenuButtonA as HTMLElement);
+    const { queryByTestId } = renderSubMenu();
 
     // TODO: Fix redundant rendering in `Menu`. The submenu is closed on initial render, but opens on second render
     // https://jira.mongodb.org/browse/LG-2904
     waitFor(async () => {
-      const subMenuItem = queryByTestId('sub-menu-item-a');
+      const subMenuItem = queryByTestId(menuItem1Id);
```

Comment:
Should this be `subMenu1Id`, not `menuItem1Id`?

Reply:
see below